python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -228,17 +236,31 @@ def func_test(model, input_sample):
                     continue
 
                 torch.set_num_threads(default_threads)
-                if validation_data is not None and metric is not None:
+                if self._calculate_accuracy:
                     result_map[method]["accuracy"] =\
                         _accuracy_calculate_helper(acce_model,
                                                    metric, validation_data)
-
+                else:
+                    result_map[method]["accuracy"] = None
                 result_map[method]["model"] = acce_model
 
             else:
                 pass
 
+        # restore model's state
+        if training_state:
+            model.train()
```

Comment:
should we consider if `self.optimized_model_dict["original"]["model"]` be set to training model? Since we directedly set model without deepcopy https://github.com/intel-analytics/BigDL/blob/b455568a73ca92683bd402c2a617bd15a40ab775/python/nano/src/bigdl/nano/pytorch/inference/optimizer.py#L183 that's not good for users.

Reply:
Good catch! Add deepcopy. Now only the original model will be set to training mode(if it is in training mode before optimize) and all accelerated model(include "original") will be set in eval mode.